### DISCUSSION TITLE
Trouble using CosineSimilarityAtThresholds from the Splink comparison library

### DISCUSSION THREAD

**Discussion by keeferaman2** [moj-analytical-services/splink#2508] [Q&A] [answered]
Hi there. 

I am trying to use the "pre-baked" Cosine Similarity comparison function, but I'm having trouble. All resources I have read around the topic seem to suggest that `CosineSimilarityAtThresholds` should work right out of the box, but it's not recognized for me. Any help on this would be greatly appreciated!

![Screenshot 2024-11-12 at 11 18 14 AM](https://github.com/user-attachments/assets/6bf00c9e-a06d-4820-bf00-46b34f7612e6)

**Comment by RobinL** [comment]
What error message do you get?

```
from splink.comparison_library import CosineSimilarityAtThresholds
CosineSimilarityAtThresholds("description")
```
works for me

The `description` field should be an array of floats (i.e. you should have converted the columns to embeddings during preprocessing, we can't do that for you)

**Reply by keeferaman2** [reply]
The error message I get is: 

`AttributeError: module 'splink.comparison_library' has no attribute 'CosineSimilarityAtThresholds'`

my Splink version is 4.0.1

And noted, thank you. I'll make sure to pass in the embeddings themselves.

**Reply by RobinL** [reply, marked as answer]
You need to upgrade to the latest version of Splink (4.0.5)

**Reply by keeferaman2** [reply]
Ah, easy enough. Thank you @RobinL
